### PR TITLE
fix: remove incorrect module path resolution, set correct rollup bundle input

### PR DESCRIPTION
**Pull request by digitalsadhu**
I'm not sure how we got here but ill say what I thinks going wrong and we can discuss solutions...

1. Using require.resolve on `@lit-labs/ssr-client/lit-element-hydrate-support.js` results in the wrong version of `lit-element-hydrate-support.js` because Lit uses conditions on their exports and it picks the `node` condition which is not actually what we want right now. We want a browser version which is called `default`. See https://github.com/lit/lit/blob/c7922a0cb90075a9e4c72f93078e411a303c54d1/packages/labs/ssr-client/package.json#L31
2. Somehow... `const input = filepath.replace('-entrypoint', '');` appears to just be totally wrong. It's rewrites away from the correct entry point to the wrong one. This makes me super suspicious. Like I must have done this for a reason... but it's really wrong. 

Solutions
1. The PR actually does seem to fix the issue... but a bunch of tests break because I'm not using an absolute URL any more and I guess pathing is off in the test folders. It's probably safer to use an absolute URL here but we would add a fair amount of hackery to get the correct file with the correct condition. Maybe there's a way of tricking Node into thinking we are in a browser context. See https://nodejs.org/api/packages.html You could run an exact and run the command using node with the flag --conditions=default. Something like `node --conditions=default index.js ` Bluurch.
2. I'm super suspicious of number 2. Probably need to go through the tests closely and look for any issues with removing the file path.replace call. Maybe add more tests to verify.

EDIT: The scope crept here and this PR now incorporates all the latest scope and strategy changes in Podium.
Tests are once again passing and assets have mostly been moved into the manifest file rather than included in the html markup with the rest of the podlet markup.

### CODE REVIEW

File: api/build.js [hunk 1]
```diff
@@ -135,6 +135,29 @@ export async function build({ state, config, cwd = process.cwd() }) {
       sourcemap: false,
     });
 
+    // build hydration support script
+    await esbuild.build({
+      entryPoints: [
+
+        // TODO: can't use require.resolve here as it will resolve in a Node context and get the wrong script
+        // probably need to create a file that does import "@lit-labs/ssr-client/lit-element-hydrate-support.js";
+        // and use that as input to entryPoints
```
Comment (wkillerud) [inline api/build.js:144]: An option might be to replace the parts of the path returned by `require.resolve` to look like what we need. Looks like from `package.json` that a `.replace("/node", "")` could do the trick. Maybe a bit brittle though.

Comment (digitalsadhu) [reply api/build.js:144]: I'm also wondering if there's not some way to tell require.resolve that we are in a browser context or something. Esbuild and Rollup get it right when you use those so there's maybe something to that though I've no idea how yet

File: api/build.js [hunk 2]
```diff
@@ -59,21 +59,22 @@ export async function build({ state, config, cwd = process.cwd() }) {
...
+        : '';
+
     // Create entrypoints for each file type
     if (existsSync(CONTENT_SRC_FILEPATH)) {
       writeFileSync(
         CONTENT_ENTRY,
-        `import "${require.resolve(
-          '@lit-labs/ssr-client/lit-element-hydrate-support.js',
-        )}";import Component from "${CONTENT_SRC_FILEPATH}";customElements.define("${NAME}-content",Component);`,
+        `${hydrateSupport}import Component from "${CONTENT_SRC_FILEPATH}";customElements.define("${NAME}-content",Component);`,
```
Comment (digitalsadhu) [inline api/build.js:71]: we can't use require.resolve here as node resolves this as a node dependency and lit provides a different dep for node vs the browser. We want the browser one.

File: plugins/lazy.js [hunk 3]
```diff
@@ -3,35 +3,21 @@ import { joinURLPathSegments } from '../lib/utils.js';
...
-                // if no document, inject at the end of the payload
-                newPayload = `${payload}<script type="module">addEventListener('load',()=>import('${url}'))</script>`;
-              }
-            }
-          }
-          done(null, newPayload);
-        },
-      );
+      // @ts-ignore
+      fastify.scriptsList.push({
```
Comment (digitalsadhu) [inline plugins/lazy.js:16]: these scriptsList.push calls are used to gather up assets for later use to do a podlet.js() call in the podlet plugin

File: plugins/podlet.js [hunk 4]
```diff
@@ -78,5 +78,20 @@ export default fp(
...
+      fastify.decorate('scriptsList', []);
+    }
+
+    // process metrics streams at the end
+    fastify.addHook('onReady', async () => {
+      // @ts-ignore
+      if (!fastify.scriptsList) return;
+
+      // @ts-ignore
+      podlet.js(fastify.scriptsList);
```
Comment (digitalsadhu) [inline plugins/podlet.js:94]: once all the other plugins have had a chance to add scripts, podlet.js() is called to register them in the manifest

File: test/plugins/plugins-mode-support.test.js [hunk 5]
```diff
@@ -55,11 +55,9 @@ test('server rendering of a lit element with hydration support', async (t) => {
     development: true,
   });
   await app.importElement(join(tmp, 'element.js'));
-  const result = app.hydrate(
```
Comment (wkillerud) [inline test/plugins/plugins-mode-support.test.js:58]: Are we losing the test for `hydrate` mode, or is it covered by the general `serverRender`?

**Review by wkillerud — APPROVED** [review]
Not sure I understand it all, but looks allright to me 👍

**Pull request merged**